src/systems/navsat/NavSat.cc

```diff
@@ -0,0 +1,266 @@
+  public: bool initialized = false;
+
+  /// \brief Create sensors in ign-sensors
+  /// \param[in] _ecm Mutable reference to ECM.
+  public: void CreateSensors(EntityComponentManager &_ecm);
+
+  /// \brief Update sensor data based on physics data
+  /// \param[in] _ecm Immutable reference to ECM.
+  public: void Update(const EntityComponentManager &_ecm);
+
+  /// \brief Remove sensors if their entities have been removed from simulation.
+  /// \param[in] _ecm Immutable reference to ECM.
+  public: void RemoveSensors(const EntityComponentManager &_ecm);
+
+  /// \brief Create sensor
+  /// \param[in] _ecm Mutable reference to ECM.
+  /// \param[in] _entity Sensor entity
+  /// \param[in] _navsat NavSat component.
+  /// \param[in] _parent Parent entity component.
+  public: void AddSensor(
+    EntityComponentManager &_ecm,
+    const Entity _entity,
+    const components::NavSat *_navSat,
+    const components::ParentEntity *_parent);
+};
+
+//////////////////////////////////////////////////
+NavSat::NavSat() : System(), dataPtr(utils::MakeUniqueImpl<Implementation>())
+{
+}
+
+//////////////////////////////////////////////////
+void NavSat::PreUpdate(const UpdateInfo &/*_info*/,
+    EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("NavSat::PreUpdate");
+  this->dataPtr->CreateSensors(_ecm);
+}
+
+//////////////////////////////////////////////////
+void NavSat::PostUpdate(const UpdateInfo &_info,
+                           const EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("NavSat::PostUpdate");
+
+  // \TODO(anyone) Support rewind
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    ignwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
+        << "s]. System may not work properly." << std::endl;
+  }
+
+  // Only update and publish if not paused.
+  if (!_info.paused)
+  {
+    this->dataPtr->Update(_ecm);
+
+    for (auto &it : this->dataPtr->entitySensorMap)
+    {
+      // Update measurement time
+      auto time = math::durationToSecNsec(_info.simTime);
+      dynamic_cast<sensors::Sensor *>(it.second.get())->Update(
+          math::secNsecToDuration(time.first, time.second), false);
+    }
+  }
+
+  this->dataPtr->RemoveSensors(_ecm);
+}
+
+//////////////////////////////////////////////////
+void NavSat::Implementation::AddSensor(
+  EntityComponentManager &_ecm,
+  const Entity _entity,
+  const components::NavSat *_navsat,
+  const components::ParentEntity *_parent)
+{
+  // create sensor
+  std::string sensorScopedName =
+      removeParentScope(scopedName(_entity, _ecm, "::", false), "::");
+  sdf::Sensor data = _navsat->Data();
+  data.SetName(sensorScopedName);
+  // check topic
+  if (data.Topic().empty())
+  {
+    std::string topic = scopedName(_entity, _ecm) + "/navsat";
+    data.SetTopic(topic);
+  }
+  std::unique_ptr<sensors::NavSatSensor> sensor =
+      this->sensorFactory.CreateSensor<
+      sensors::NavSatSensor>(data);
```

Comment:
```suggestion
  std::unique_ptr<sensors::NavSatSensor> sensor =
      this->sensorFactory.CreateSensor<sensors::NavSatSensor>(data);
```
nit: place the template on the same line for readability